### PR TITLE
Fix solution selection

### PR DESCRIPTION
**Pull request by BarryThePenguin**
Fixes selection of solutions

Before
Screen Shot 2020-11-02 at 10 15 51 pm

After
Screen Shot 2020-11-02 at 10 18 30 pm

Allows fixing individual errors by clicking "Fix"

I also removed the usage of `waitsForPromise`

### CODE REVIEW

File: lib/format.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const ruleURI = require('eslint-rule-documentation');
 
 // (editor: Object, message: Object) => Array<Object>
 function selectSolutions(editor, message) {
-	const fix = {message};
+	const {fix} = message;
```
Comment (BarryThePenguin) [inline lib/format.js:7]: Messages have a `fix` property. This was creating a `fix` object with a `message` property

And https://eslint.org/docs/developer-guide/nodejs-api#-lintmessage-type

File: lib/format.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ function selectSolutions(editor, message) {
 	}
 
 	const position = new Range(...range.map(index => buffer.positionForCharacterIndex(index)));
-	return {position, replaceWith: text};
+	return [{position, replaceWith: text}];
```
Comment (BarryThePenguin) [inline lib/format.js:25]: Solutions on lint messages should be an array

See https://github.com/steelbrain/linter/blob/master/docs/types/linter-message-v2.md

File: lib/format.test.js [hunk 3]
```diff
@@ -77,3 +77,30 @@ test('for a report with an error message with a location', t => {
...
+	const [{solutions: [actual]}] = format(editor)(input);
+	const expected = {
+		position: {
+			end: {column: 0, row: 0},
+			start: {column: 0, row: 0}
+		},
+		replaceWith: ';'
+	};
+	t.like(actual, expected, 'it should return the correct location');
+});
```
Comment (BarryThePenguin) [inline lib/format.test.js:106]: Unsure how useful a test like this is.. I think the additional assertions to the spec are fine

File: spec/linter-xo-spec.js [hunk 4]
```diff
@@ -17,107 +16,94 @@ describe('xo provider for linter', () => {
...
+			const [message] = await lint(editor);
+			expect(message.location.file).toEqual(files.bad);
+			expect(message.severity).toEqual('error');
+			expect(message.excerpt).toEqual('Strings must use singlequote.');
+			expect(message.url).toEqual('https://eslint.org/docs/rules/quotes');
+
+			expect(message.location.position).toEqual([[1, 12], [1, 17]]);
+			expect(message.solutions.length).toBe(1);
+			expect(message.solutions[0].position).toEqual({start: {row: 1, column: 12}, end: {row: 1, column: 17}});
+			expect(message.solutions[0].replaceWith).toBe('\'bar\'');
```
Comment (BarryThePenguin) [inline spec/linter-xo-spec.js:42]: These are the new assertions for the fix to `solutions`

**Comment by sindresorhus** [comment]
Thanks for fixing this. Do you plan more PRs here or should I do a new release?

**Comment by BarryThePenguin** [comment]
No new PRs planned at this stage. I'm still investigating #104

**Pull request merged**